hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/googlepubsub/MessageCompressor.java

```diff
@@ -17,15 +17,18 @@ byte[] compress(byte[] data) throws IOException {
         Codec codec = codecFactory.createInstance();
 
         ByteBuffer compressed = codec.compress(ByteBuffer.wrap(data));
+        byte[] compressedBytes = new byte[compressed.limit()];
+        compressed.get(compressedBytes);
```

Comment:
This can potentially cause some CPU increase, look at that if there will be a problem

Reply:
so previously we used array which was backing the `compressed` ByteBuffer but now we have to copy the underlying array to the new array. Why? Could you provide more context what changed?

Reply:
I believe we can use ByteBuffer directly instead of copying the underlying array, please see: https://github.com/allegro/hermes/pull/1854

Reply:
The previous version of the compressor returned `array` with pure data, and we used that as a source for pubsub data sender. The newer version of compressor is using some kind of buffer partially filled with data. Because we returned the array directly from compressed ByteBuffer it contained a lot of empty cells, so it couldn't be directly returned to the Pubsub sender.

Reply:
I didn't notice the API accepting the ByteBuffer ifself, so your change will work, thank you 👍